src/main/java/com/google/sps/model/GmailClientImpl.java

```diff
@@ -42,13 +48,42 @@ private GmailClientImpl(Credential credential) {
             .build();
   }
 
+  /**
+   * List the messages in a user's Gmail account that match the passed query. Results are returned
+   * in batches of 100 from the Gmail API, though a next page token is included if another batch
+   * still exists. This method will use these tokens to get all of the emails that match the query,
+   * even if that is in excess of 100.
```

Comment:
It's weird to have javadoc on a method override, because the external API of the method should be identical to the original declaration. Consider adding a comment with these implementation details inside the method body.

Reply:
Good point - Thanks for this! Fixed